dgl_sparse/src/cpu/matrix_ops_impl.cc

```diff
@@ -0,0 +1,69 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ * @file cpu/matrix_ops_impl.cc
+ * @brief DGL CPU implementation of matrix operators.
+ */
+#include "../matrix_ops_impl.h"
+
+#include "../utils.h"
+
+namespace dgl {
+namespace sparse {
+
+template <c10::DeviceType XPU, typename IdType>
+std::tuple<std::shared_ptr<COO>, torch::Tensor, torch::Tensor>
+COOIntersectionImpl(
+    const std::shared_ptr<COO> &lhs, const std::shared_ptr<COO> &rhs) {
+  const int64_t lhs_len = lhs->indices.size(1);
+  const int64_t rhs_len = rhs->indices.size(1);
+
+  torch::Tensor lhs_row = lhs->indices.index({0}).contiguous();
+  const IdType *lhs_row_ptr = static_cast<IdType *>(lhs_row.data_ptr());
```

Comment:
Could provide a utility function to get data pointers, e.g., `const IdType *lhs_row_ptr = utils::GetPtr<IdType>(lhs->indices.index({0}))`.